src/platform/backends/qemu/qemu_virtual_machine.cpp

```diff
@@ -670,6 +628,63 @@ void mp::QemuVirtualMachine::initialize_vm_process()
     });
 }
 
+void mp::QemuVirtualMachine::connect_signals()
+{
+    auto connection_type = static_cast<Qt::ConnectionType>(Qt::QueuedConnection | Qt::UniqueConnection);
```

Comment:
Check the note here https://doc.qt.io/qt-6/qobject.html#connect stating that "Qt::UniqueConnections do not work for lambdas, non-member functions and functors; they only apply to connecting to member functions.". I did an experiment and indeed the connections are not valid. Unfortunately you're going to need a different mechanism for this :neutral_face: